datasets/mdd/mdd.py

```diff
@@ -0,0 +1,569 @@
+    the use by either party of the trademark "Creative Commons" or any
+    related trademark or logo of Creative Commons without the prior
+    written consent of Creative Commons. Any permitted use will be in
+    compliance with Creative Commons' then-current trademark usage
+    guidelines, as may be published on its website or otherwise made
+    available upon request from time to time. For the avoidance of doubt,
+    this trademark restriction does not form part of this License.
+
+    Creative Commons may be contacted at https://creativecommons.org/.
+"""
+
+ZIP_URL = "http://www.thespermwhale.com/jaseweston/babi/movie_dialog_dataset.tgz"
+REDDIT_URL = "http://tinyurl.com/p6tyohj"
+dir = "movie_dialog_dataset/"
+dir2 = ""
+paths = {
+    "task1_qa": {
+        "train": dir + "task1_qa/task1_qa_train.txt",
+        "dev": dir + "task1_qa/task1_qa_dev.txt",
+        "test": dir + "task1_qa/task1_qa_test.txt",
+    },
+    "task2_recs": {
+        "train": dir + "task2_recs/task2_recs_train.txt",
+        "dev": dir + "task2_recs/task2_recs_dev.txt",
+        "test": dir + "task2_recs/task2_recs_test.txt",
+    },
+    "task3_qarecs": {
+        "train": dir + "task3_qarecs/task3_qarecs_train.txt",
+        "dev": dir + "task3_qarecs/task3_qarecs_dev.txt",
+        "test": dir + "task3_qarecs/task3_qarecs_test.txt",
+    },
+    "task4_reddit": {
+        "train": "task4_reddit/task4_reddit_train.txt",
+        "dev": "task4_reddit/task4_reddit_dev.txt",
+        "test": "task4_reddit/task4_reddit_test.txt",
+        "cand_valid": "task4_reddit/task4_reddit_cand-valid.txt",
+        "cand_test": "task4_reddit/task4_reddit_cand-test.txt",
+    },
+}
+
+
+class Mdd(datasets.GeneratorBasedBuilder):
+    """The Movie Dialog Dataset"""
+
+    VERSION = datasets.Version("1.1.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="task1_qa", version=VERSION, description="This part of my dataset covers task1_qa part of the dataset"
+        ),
+        datasets.BuilderConfig(
+            name="task2_recs",
+            version=VERSION,
+            description="This part of my dataset covers task2_recs part of the dataset",
+        ),
+        datasets.BuilderConfig(
+            name="task3_qarecs",
+            version=VERSION,
+            description="This part of my dataset covers task3_qarecs part of the dataset",
+        ),
+        datasets.BuilderConfig(
+            name="task4_reddit",
+            version=VERSION,
+            description="This part of my dataset covers task4_reddit part of the dataset",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = (
+        "task1_qa"  # It's not mandatory to have a default configuration. Just use one if it make sense.
+    )
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "dialogue_turns": datasets.Sequence(
+                    {
+                        "speaker": datasets.ClassLabel(names=["0", "1"]),
```

Comment:
You can just have an integer here I think, "speaker" is not a label we want to predict